docs/developer/architecture/configurations.rst

```diff
@@ -91,17 +91,45 @@ requirements for the following reasons:
 How it works
 ^^^^^^^^^^^^
 
-During the Bootstrap stages, when we are assertive that the K8s cluster is
-fully ready and available we could perform the following actions:
+During Bootstrap, Upgrade or Downgrade stages, when we are assertive that
+the K8s cluster is fully ready and available we could perform the following
+actions:
 
-  - Create and deploy ConfigMaps that hold cluster and service configurations
-    and pre-fill them with default values.
-  - Template service pods and deployments to consume configuration data
-    directly from the above deployed ConfigMaps
+  - Firstly, create and deploy ConfigMaps that will hold customizable cluster
+    and service configurations.
+    These ConfigMaps should define an empty `config.yaml` in the data section
+    of the ConfigMap for later use. A standard layout for each customizable
+    field could be added in the documentation to assist MetalK8s administrator
+    in adding and modifying customizations.
+
+  - In a `csc.yaml` file located at the root directory of each MetalK8s Addon,
+    define the keys and values for default service configurations in a
+    YAML structured format.
+
+      - The layout of service configurations within this file could have root
+        key being the impacted ConfigMap name. For example:
+
+    .. code-block:: yaml
+
+        metalk8s-prometheus-config:
+          spec:
+            deployment:
+              replicas: 1
```

Comment:
So, as mentioned above, I'd rather have a single "config" in those `csc.yaml` files (I don't like the name, will suggest another below), and make it exhaustive (if some value **must not** have a default, still provide an example as a comment). For instance, the `salt/metalk8s/addons/prometheus-operator/config/grafana.yaml`:

```
# Configuration of the Grafana service
apiVersion: addons.metalk8s.scality.com/v1alpha1
kind: GrafanaConfig
spec:
  # Configure the Grafana Deployment
  deployment:
    replicas: 1
```

Reply:
Let's add an item for having these files included in the docs. That'll make things easier on our side.